core/evm.go

```diff
@@ -39,7 +40,7 @@ type ChainContext interface {
 }
 
 // NewEVMBlockContext creates a new context for use in the EVM.
-func NewEVMBlockContext(header *types.Header, chain ChainContext, author *common.Address) vm.BlockContext {
+func NewEVMBlockContext(header *types.Header, chain ChainContext, config *params.ChainConfig, author *common.Address) vm.BlockContext {
```

Comment:
I'm wondering whether it makes sense to include the config in the `ChainContext` interface? It seems like most callers are calling this with a blockchain or headerchain object anway

Reply:
Felix also mentioned this. Originally when I implemented it, I ran into a couple issues, but I gave it another shot and I think was able to pull it off in [b16db48](https://github.com/ethereum/go-ethereum/pull/31002/commits/b16db487894f08ca39f6c1f0824e7c13a17227c9).

This approach definitely seems nicer.